addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/convention/homeassistant/BaseChannelConfiguration.java

```diff
@@ -29,59 +29,59 @@
  * @author Jochen Klein - Initial contribution
  */
 @NonNullByDefault
-public abstract class HAConfiguration {
-
-    public String name;
-
-    protected String icon = "";
-    protected int qos = 0;
-    protected boolean retain = false;
-    protected @Nullable String value_template;
-    protected @Nullable String unique_id;
-
-    protected @Nullable String availability_topic;
-    protected String payload_available = "online";
-    protected String payload_not_available = "offline";
-
-    @SerializedName(value = "~")
-    protected String tilde = "";
+public abstract class BaseChannelConfiguration {
 
     /**
      * This class is needed, to be able to parse only the common base attributes.
-     * Without this, {@link HAConfiguration} cannot be instantiated, as it is abstract.
+     * Without this, {@link BaseChannelConfiguration} cannot be instantiated, as it is abstract.
      * This is needed during the discovery.
      */
-    private static class Config extends HAConfiguration {
+    private static class Config extends BaseChannelConfiguration {
         public Config() {
             super("private");
         }
     }
 
     /**
-     * Parse the configJSON into a subclass of {@link HAConfiguration}
+     * Parse the configJSON into a subclass of {@link BaseChannelConfiguration}
      *
      * @param configJSON
      * @param gson
      * @param clazz
      * @return configuration object
      */
-    public static <C extends HAConfiguration> C fromString(final String configJSON, final Gson gson,
+    public static <C extends BaseChannelConfiguration> C fromString(final String configJSON, final Gson gson,
             final Class<C> clazz) {
         return gson.fromJson(configJSON, clazz);
     }
 
     /**
-     * Parse the base properties of the configJSON into a {@link HAConfiguration}
+     * Parse the base properties of the configJSON into a {@link BaseChannelConfiguration}
      *
      * @param configJSON
      * @param gson
      * @return configuration object
      */
-    public static HAConfiguration fromString(final String configJSON, final Gson gson) {
+    public static BaseChannelConfiguration fromString(final String configJSON, final Gson gson) {
         return fromString(configJSON, gson, Config.class);
     }
 
-    protected HAConfiguration(String defaultName) {
+    public String name;
+
+    protected String icon = "";
+    protected int qos = 0;
```

Comment:
For `int = 0` and `boolean ? false` you do not have to assign them. They are default.